### ISSUE TITLE
[Feature Request] Allow easier custom goto-mode key binding

### ISSUE THREAD

**Issue by yuchenshi** [71/dance#86]
Say I want to implement a custom "g x" binding to trigger some jumping using a command provided by some other extension. Right now, if I add a "g x" binding in my `keybindings.json`, Dance `g j` for example will stop working, because VSCode will wait for the second key when `g` is pressed instead of showing the popup menu. And once `j` is pressed next, VSCode will complain that `g j` is not a valid binding.

As a user, I think my only way to make it work will be to create one entry for each of the builtin "g _" command in my `keybindings.json` and somehow re-implement each one using a custom extension. This sucks and I think Dance can make it better by actually make "goto" a mode, so I can do something like:

```json
  {
    "key": "x",
    "command": "someRandomExtension.someCommand",
    "when": "editorTextFocus && dance.mode == 'goto'" // <-- new mode called "goto"
  },
```

Builtin `g j` etc. can also be re-implemented this way so it plays nicely with the rest of the keybindings (e.g. conditionally overrideable).

The same argument can be made for other commands that prompts a key, and view-mode commands (#40) if Dance ever supports them.

(Other implementation ideas may include overwriting the "type" command (or only overwriting when `g` is pressed) but those seems more intrusive.)

**Comment by 71** [comment]
This is one way of doing things that integrates nicely with VS Code, but with one significant problem: the UI we display on `goto` will no longer be complete. I think I'd rather add a command for each `goto` subcommand, and use user preferences to customize that menu (and hell, even allow arbitrary custom menus) with a list like 
```json
"dance.menus": {
  "goto": {
    "[key]": {
      "command": "...",
      "description": "...",
    },
  },
}
```

And then:

```json
{ "key": "g",
  "command": "dance.showMenu",
  "args": {
    "menu": "goto",
  },
}
```

It would be easier to define and it would show a complete dialog. 

Thoughts?

**Comment by 71** [comment]
Now implemented in `unstable`: 1777a2893a9f1d68aa7fa2563f7a7b4b98c1a846.

The only change compared to the snipped [above](#issuecomment-610006062) is that `description` is renamed to `text`. Not sure that's better, though.
